tests/integration/targets/aci_domain_to_encap_pool/tasks/main.yml

```diff
@@ -0,0 +1,620 @@
+    pool_allocation_mode: dynamic
+    vm_provider: kubernetes
+    state: absent
+  register: remove_vmm_kubernetes
+
+- name: Remove vmm microsoft
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    domain: vmm_dom
+    pool: test_pool
+    pool_type: vlan
+    pool_allocation_mode: dynamic
+    vm_provider: microsoft
+    state: absent
+  register: remove_vmm_microsoft
+
+- name: Remove vmm redhat
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    domain: vmm_dom
+    pool: test_pool
+    pool_type: vlan
+    pool_allocation_mode: dynamic
+    vm_provider: redhat
+    state: absent
+  register: remove_vmm_redhat
+
+- name: Remove vmm openstack
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    domain: vmm_dom
+    pool: test_pool
+    pool_type: vlan
+    pool_allocation_mode: dynamic
+    vm_provider: openstack
+    state: absent
+  register: remove_vmm_openstack
+
+- name: Remove vmm openshift
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    domain: vmm_dom
+    pool: test_pool
+    pool_type: vlan
+    pool_allocation_mode: dynamic
+    vm_provider: openshift
+    state: absent
+  register: remove_vmm_openshift
+
+- name: Remove vmm vmware
+  aci_domain_to_encap_pool:
+    <<: *aci_info 
+    domain_type: vmm
+    domain: vmm_dom
+    pool: test_pool
+    pool_type: vlan
+    pool_allocation_mode: dynamic
+    vm_provider: vmware
+    state: absent
+  register: remove_vmm_vmware
+
+- name: Verify Remove
+  assert:
+    that:
+    - remove_phys.previous.0.physDomP.children.0.infraRsVlanNs.attributes.tDn =='uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_fc.previous.0.fcDomP.children.0.fcRsVsanNs.attributes.tDn == 'uni/infra/vsanns-[test_pool]-dynamic'
+    - remove_l2dom.previous.0.l2extDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-static'
+    - remove_l3dom.previous.0.l3extDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-static'
+    - remove_vmm_cloudfoundry.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_kubernetes.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_microsoft.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_openshift.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_openstack.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_redhat.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+    - remove_vmm_vmware.previous.0.vmmDomP.children.0.infraRsVlanNs.attributes.tDn == 'uni/infra/vlanns-[test_pool]-dynamic'
+
+# Clean Environment Again
+- name: Remove physical domain
```

Comment:
Not sure we need to clean this up afterward. But if you want to do it again, use a loop like above.